tests/manual-test-cases/Group5-Functional-Tests/5-22-NFS-Volume.robot

```diff
@@ -0,0 +1,249 @@
+    :FOR  ${id}  IN  @{containers}
+    \   ${rc}  ${stopOutput}=  Run And Return Rc And Output  docker %{VCH-PARAMS} stop ${id}
+    \   Should Be Equal As Integers  ${rc}  0
+
+
+Simple docker volume inspect
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} volume inspect ${nfsNamedVolume}
+    Should Be Equal As Integers  ${rc}  0
+    ${output}=  Evaluate  json.loads(r'''${output}''')  json
+    ${id}=  Get From Dictionary  ${output[0]}  Name
+    Should Be Equal As Strings  ${id}  ${nfsNamedVolume}
+
+Simple Volume ls test
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} volume ls
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${output}  vsphere
+    Should Contain  ${output}  ${nfsNamedVolume}
+    Should Contain  ${output}  ${nfsUnNamedVolume}
+    Should Contain  ${output}  DRIVER
+    Should Contain  ${output}  VOLUME NAME
+
+Volume rm tests
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} volume rm ${nfsUnNamedVolume}
+    Should Be Equal As Integers  ${rc}  0
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} volume ls
+    Should Be Equal As Integers  ${rc}  0
+    Should Not Contain  ${output}  ${nfsUnNamedVolume}
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} volume rm ${nfsNamedVolume}
+    Should Be Equal As Integers  ${rc}  1
+    Should Contain  ${output}  Error response from daemon: volume ${nfsNamedVolume} in use by
+
+Kill NFS Server
+    ${rc}  ${runningContainer}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run -d -v ${nfsNamedVolume}:/mydata ${busybox} sh -c "while true; do echo 'Still here...\n' >> /mydata/test_nfs_kill.txt; sleep 1; done"
+    Should Be Equal As Integers  ${rc}  0
+
+    ${rc}  ${tailOutput}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run -v ${nfsNamedVolume}:/mydata ${busybox} sh -c "tail -5 /mydata/test_nfs_kill.txt"
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${tailOutput}  Still here...
+
+    Kill Nimbus Server  %{NIMBUS_USER}  %{NIMBUS_PASSWORD}  ${NFS}
+
+    ${rc}  ${tailOutput}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run -v ${nfsNamedVolume}:/mydata ${busybox} sh -c "tail -5 /mydata/test_nfs_kill.txt"
+    Should Be Equal As Integers  ${rc}  125
+    Should Contain  ${tailOutput}  Server error from portlayer: unable to wait for process launch status:
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run -v ${nfsNamedVolume}:/mydata ${busybox} sh -c "echo 'Where am I writing to?...\n' >> /mydata/test_nfs_kill.txt"
+    Should Be Equal As Integers  ${rc}  125
+
+    ${rc}  ${lsOutput}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run -v ${nfsNamedVolume}:/mydata ${busybox} sh -c "ls mydata"
+    Should Be Equal As Integers  ${rc}  125
+    Should Contain  ${lsOutput}  Server error from portlayer: unable to wait for process launch status:
```

Comment:
@matthewavery, please correct me if I'm wrong here. 
@mhagen-vmware  I don't think I can mount a read-only NFS volume to the VCH at install time at the moment.
I can do something like, mount a read/write NFS volume and then maybe change the permissions of the volume folder inside to read-only afterwards?  I'm not sure if that is what you were looking for in this case but the volume will be created as read/write from the beginning.

As for the error output of that last test, this is what I get:
```
docker: Error response from daemon: Server error from portlayer: unable to wait for process launch status: Post https://10.192.219.205/sdk: context deadline exceeded.
[vic-integration-test-on-pr:L86:1521s] time="2017-07-17T09:38:44Z" level=error msg="error getting events from daemon: net/http: request canceled" 
```
@mhagen-vmware , do you want me to write a bug asking for better output on that?